test/tracerEq/test_bcs_2d.py

```diff
@@ -0,0 +1,161 @@
+        I = 0
+        tau = 0
+        dt = 0.05
+        while tau < time - 0.5*dt:
+            I += exp(-nu*(n*pi/lx)**2*(t-tau))
+            tau += dt
+        I *= source_fourier_coeff(n)
+        return I*phi(n)
+
+    def ic_term(n, time):
+        return ic_fourier_coeff(n)*exp(-nu*(n*pi/lx)**2*time)*phi(n)
+
+    # Assemble truncated Fourier expansion
+    sol = Function(P1, name='Fourier expansion')
+    num_terms_source = 1  # Only one needed since source is constant
+    num_terms_ic = 100
+    expr = Constant(0.5*source_fourier_coeff(0)*time)
+    expr = expr + Constant(0.5*ic_fourier_coeff(0))
+    for k in range(1, num_terms_source):
+        expr = expr + source_term(k, time)
+    for k in range(1, num_terms_ic):
+        expr = expr + ic_term(k, time)
+    expr -= ic
+    sol.interpolate(-expr)
+
+    return sol
+
+
+def run(refinement, **model_options):
+
+    # Domain
+    lx = 10
+    ly = 1
+    nx = 40*refinement
+    ny = 4
+    mesh2d = RectangleMesh(nx, ny, lx, ly)
+    depth = 40.0
+
+    # Time interval
+    dt = 0.1/refinement
+    t_end = 1.0
+    t_export = 0.1
+
+    # Bathymetry
+    P1_2d = FunctionSpace(mesh2d, 'CG', 1)
+    bathy_2d = Function(P1_2d, name='Bathymetry')
+    bathy_2d.assign(depth)
+
+    # Diffusivity and diffusive flux BC to impose
+    nu = Constant(0.1)
+    diff_flux = 0.2
+
+    # Solver
+    solver_obj = solver2d.FlowSolver2d(mesh2d, bathy_2d)
+    options = solver_obj.options
+    options.output_directory = 'outputs'
+    options.no_exports = True
+    options.timestep = dt
+    options.simulation_end_time = t_end - 0.5*dt
+    options.simulation_export_time = t_export
+    options.solve_tracer = True
+    options.tracer_only = True
+    options.horizontal_diffusivity = nu
+    options.use_limiter_for_tracers = True
+    options.fields_to_export = ['tracer_2d']
+    options.update(model_options)
+
+    # Boundary conditions
+    solver_obj.bnd_functions['tracer'] = {1: {'diff_flux': diff_flux*nu}}
+    # NOTE: Zero diff_flux boundaries are enforced elsewhere by default
+
+    # Run model
+    solver_obj.assign_initial_conditions()
+    solver_obj.iterate()
+    sol = solver_obj.fields.tracer_2d
+
+    # Get truncated Fourier series solution
+    fsol = fourier_series_solution(mesh2d, t_end)
+    if not options.no_exports:
+        File('outputs/fourier_series_solution.pvd').write(fsol)
+
+    return errornorm(sol, fsol)
+
+
+def run_convergence(**model_options):
+    errors = []
+    for refinement in (1, 2, 4):
+        errors.append(run(refinement, **model_options))
+    msg = "Wrong convergence rate {:.4f}, expected 2.0000."
+    slope = errors[0]/errors[1]
+    assert slope > 2, msg.format(slope)
+    slope = errors[1]/errors[2]
+    assert slope > 2, msg.format(slope)
+
+
+@pytest.fixture(params=[1, 2])
+def polynomial_degree(request):
+    return request.param
+
+
+@pytest.mark.parametrize(('stepper'),
+                         [('CrankNicolson')])
```

Comment:
Does this work with other time steppers too? If possible we should test them for better code coverage (to avoid issues like #172).

Reply:
As I understand it, Crank Nicolson is the only tracer timestepper currently available within the `create_timesteppers` method of `solver2d`. None of the other 2d tracer tests use anything other than Crank Nicolson. Maybe a future PR could consider extending the options and tests?

Reply:
I see, that indeed seems to be the case. Yeah, let's address this in another issue.